pkg/dev/dev.go

```diff
@@ -206,6 +222,76 @@ func (m *Manager) SetWorkflowSecret(ctx context.Context, workflow, key, value st
 	return vm.writeSecrets(ctx, secret)
 }
 
+func (m *Manager) InitializeMetadataApi(ctx context.Context, mockOptions MetadataMockOptions) (string, error) {
```

Comment:
I would consider creating another manager type for this. Maybe something like `MockMetadataAPIManager` or something. It's my fault, but `Manager` has a lot of tight coupling with the kubernetes cluster management that's not actually required here, so you'd probably save some lines and make the the feature you're creating easier to extend in the future by moving it off this manager type. If you look at your command function in pkg/cmd/metadata.go, you can see you are required to create a lot of unnecessary dependencies that need to be injected into `Manager` in order to even create it.